ReactiveFeedback/Sources/SignalProducer+System.swift

```diff
@@ -2,29 +2,33 @@ import Foundation
 import ReactiveSwift
 import enum Result.NoError
 
-public typealias Reducer<State, Event> = (State, Event) -> State
+extension SignalProducer where Error == NoError {
+    public static func system<Event>(
+        initial: Value,
+        scheduler: Scheduler = QueueScheduler.main,
+        reduce: @escaping (Value, Event) -> Value,
+        feedbacks: [FeedbackLoop<Value, Event>]
+    ) -> SignalProducer<Value, NoError> {
+        return SignalProducer.deferred {
+            let (state, observer) = Signal<Value, NoError>.pipe()
 
-extension SignalProducerProtocol where Error == NoError {
+            let events = feedbacks.map { feedback in
+                return feedback.loop(scheduler, state)
+            }
 
-    public static func system<Event>(initialState: Value,
-                                     scheduler: Scheduler = QueueScheduler.main,
-                                     reduce: @escaping Reducer<Value, Event>,
-                                     feedback: [FeedbackLoop<Value, Event>]) -> SignalProducer<Value, NoError> {
-        return SignalProducer.deferred {
-            let (subject, observer) = Signal<Value, NoError>.pipe()
-            let events = Signal<Event, NoError>.merge(feedback.map { feedback in
-                return feedback.loop(scheduler, subject)
-            })
-            return SignalProducer(events.scan(initialState, reduce))
-                .prefix(value: initialState)
+            return SignalProducer<Event, NoError>(Signal.merge(events))
```

Comment:
Why remove `deferred`?

Reply:
Oh, my bad did not notice it =)

Reply:
yet to understand the usage of `deferred`, based on the discussions we had on Slack. ☹️